homeassistant/components/spokestack/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Spokestack TTS component."""
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS = ["tts"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Hello World component."""
+    # Ensure our name space for storing objects is a known type. A dict is
+    # common/preferred as it allows a separate instance of your class for each
+    # instance that has been created in the UI.
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Spokestack from a config entry."""
+    hass.data[DOMAIN][entry.entry_id] = entry.data["voice"]
+
+    hass.data[DOMAIN].update(entry.data)
```

Comment:
I don't get this, why not just store `entry.data` in there to begin with?

Reply:
If it is wrong then I didn't understand the correct way to do it and apologize for wasting your time. Do you mind pointing me to the developer documentation that covers the correct way to do this?

Reply:
`entry.data` is a dictionary. In L26 you store 1 piece of information in it. Then in line 28 you store all the rest. `hass.data[DOMAIN] = entry.data` would do the trick.

Reply:
ok that makes sense. thank you!